compiler error with catch

This piece of code fails to compile properly (with current master):

```
const FooError = error { Foo };

fn foo() FooError!void {
    return FooError.Foo;
}

pub fn main() void {
    foo() catch 0;
}

```
It fails with:
```
% zig build-exe test.zig
Assertion failed at /home/szekeres/.cache/yay/zig-git/src/zig/src/analyze.cpp:492
   in get_pointer_to_type_extra2. This is a bug in the Zig compiler.
Unable to dump stack trace: debug info stripped
zsh: abort (core dumped)  zig build-exe test.zig
```

The stacktrace from 0.5.0
```
zig run /home/mgxm/zig/issues/3703/src/main.zig
/home/mgxm/zig/issues/3703/src/main.zig:10:17: error: integer value 0 cannot be implicitly casted to type 'void'
    foo() catch 0;
                ^
/home/mgxm/zig/issues/3703/src/main.zig:10:11: note: referenced here
    foo() catch 0;
          ^
```

--- a/src/ir.cpp
+++ b/src/ir.cpp
@@ -17695,6 +17695,8 @@ static IrInstruction *ir_analyze_instruction_phi(IrAnalyze *ira, IrInstructionPh
             peer_parent->resolved_type = ir_resolve_peer_types(ira,
                     peer_parent->base.source_instruction->source_node, expected_type, instructions,
                     peer_parent->peers.length);
+            if (type_is_invalid(peer_parent->resolved_type))
+                return ira->codegen->invalid_instruction;
 
             // the logic below assumes there are no instructions in the new current basic block yet
             ir_assert(ira->new_irb.current_basic_block->instruction_list.length == 0, &phi_instruction->base);

--- a/test/compile_errors.zig
+++ b/test/compile_errors.zig
@@ -3,6 +3,18 @@ const builtin = @import("builtin");
 
 pub fn addCases(cases: *tests.CompileErrorContext) void {
     cases.add(
+        "missing result type for phi node",
+        \\fn foo() !void {
+        \\    return anyerror.Foo;
+        \\}
+        \\export fn entry() void {
+        \\    foo() catch 0;
+        \\}
+    ,
+        "tmp.zig:5:17: error: integer value 0 cannot be coerced to type 'void'",
+    );
+
+    cases.add(
         "atomicrmw with enum op not .Xchg",
         \\export fn entry() void {
         \\    const E = enum(u8) {

